xlators/mgmt/glusterd/src/glusterd-op-sm.c

```diff
@@ -2830,6 +2832,151 @@ glusterd_set_shared_storage(dict_t *dict, char *key, char *value,
+    else
+        return ret;
+}
+
+// Compare values in the set in the volume/cluster with that of the predicate
+
+int
+glusterd_compare_option_value(glusterd_volinfo_t *volinfo, char *input_key,
+                              int input_value)
+{
+    struct volopt_map_entry *vme = NULL;
+    int ret = -1;
+    char *deflt_val = NULL;
+    int keylen;
+    glusterd_conf_t *priv = NULL;
+    xlator_t *this = THIS;
+
+    priv = this->private;
+    GF_VALIDATE_OR_GOTO(this->name, priv, out);
+
+    for (vme = &glusterd_volopt_map[0]; vme->key; vme++) {
+        if (!strcmp(vme->key, input_key)) {
+            // First look for the key in the priv->opts for global option
+            keylen = strlen(vme->key);
+            ret = dict_get_strn(priv->opts, vme->key, keylen, &deflt_val);
+            if (!deflt_val) {
+                // Check for the option in the volinfo dict
+                ret = dict_get_strn(volinfo->dict, vme->key, keylen,
+                                    &deflt_val);
+                if (ret == -ENOENT)
+                    // Check for glusterd default values
+                    deflt_val = glusterd_get_option_value(volinfo, vme->key);
+                if (!deflt_val) {
+                    if (vme->value) {
+                        deflt_val = vme->value;
+                    } else {
+                        // Check the option for tranlator default value
+                        ret = glusterd_get_value_for_vme_entry(vme, &deflt_val);
+                        if (ret)
+                            goto out;
+                        else if (ret == -2)
+                            continue;
+                    }
+                }
+            }
+        }
+    }
+
+    ret = glusterd_compare_bool_value(deflt_val, input_value);
+    deflt_val = NULL;
+
+out:
+    if (deflt_val)
+        GF_FREE(deflt_val);
+
+    return ret;
+}
+
+// Check the options set in the cluster/per-volume satisfy the dependency chain
+
+int
+glusterd_dependency_chain_check(glusterd_volinfo_t *volinfo, char *key,
+                                char *value, char **errstr)
+{
+    int ind = 0;
+    int j = 0;
+    int match_cnt = 0;
+    int ret = -1;
+    char *err_s = NULL;
+
+    // Going through each of the predicates in the prohibhited_clauses list and
+    // comparing the values for each of the options for a volume
+    while (prohibited_clauses[ind]) {
+        // Check if there are any predicates left in the list
+        if (!prohibited_clauses[ind][0].op)
+            break;
+
+        // Copy the predicate clause, which should be logged in case of failure
+        if (prohibited_clauses[ind][0].val == CLAUSE_LABEL) {
+            err_s = gf_strdup(prohibited_clauses[ind][0].op);
+        } else {
+            match_cnt = 0;
+            j = 0;
+            while (j >= 0) {
+                // Check the end of a list of options for single predicate
+                if (!prohibited_clauses[ind][j].op)
+                    break;
```

Comment:
Same as above. `j` starts as `0` and just gets increased. The `while` condition will never fail. Drop the `if`` statement and advance its condition to the position of the `while` exit condition.